src/urllib3/_collections.py

```diff
@@ -12,7 +14,16 @@ def __exit__(self, exc_type, exc_value, traceback):
             pass
 
 
-from collections import OrderedDict
+# Starting in Python 3.7 the 'dict' class is guaranteed to be
+# ordered by insertion. This behavior was an implementation
+# detail in Python 3.6. OrderedDict is implemented in C in
```

Comment:
And crucially it's also true in PyPy.

Why can't we also use `dict` in Python 3.6 by the way? Is this for other implementations not in our CI that might not have the properties we want?

Reply:
@pquentin Yeah basically! Being conservative :)